.github/workflows/tag-build.yaml

```diff
@@ -1,10 +1,220 @@
+        run: |
+          # Do nothing
+  # Each section here could be split out into a separate job, at the cost of slightly increased complexity.
+  # This would improve the (already somewhat fast) performance a bit, but I'm not sure if it's worth the
+  # tradeoff.
+  build-plugins:
+    needs:
+      - setup
+    runs-on: ubuntu-22.04-32core
+    environment: ${{ needs.setup.outputs.environment }}
+    permissions:
+      contents: read
+      id-token: write
+    env:
+      ARTIFACT_DIRECTORY: /tmp/build
+    steps:
+      # Setup
+      - name: Enable performance telemetry/metrics
+        uses: catchpoint/workflow-telemetry-action@v1
+        with:
+          comment_on_pr: false
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.setup.outputs.gitref }}
+          fetch-depth: 0  # This is required by some of the commands in the makefiles
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "./go.mod"
+          cache-dependency-path: "./go.sum"
+          check-latest: true
+      - name: Set environment variables for Makefiles
+        env:
+          VERSION_TAG: ${{ needs.setup.outputs.version }}
+        run: |
+          {
+            echo "VERSION=${VERSION_TAG##v}"
+            echo "GITREF=$VERSION_TAG" >> "$GITHUB_ENV"
+            echo "GNUMAKEFLAGS=-j$(nproc)"
+          } >> "$GITHUB_ENV"
+      # File artifacts
+      - name: Build the release tarballs
+        run: |
+          # Binaries and Helm charts
+          make releases helm-package-charts
+
+          # Terraform provider and event handler, as appropriate
+          go install github.com/konoui/lipo@latest  # At some point this should be merged into the buildbox
+          make OS=linux ARCH=amd64 release/terraform release/event-handler
+          make OS=linux ARCH=arm64 release/terraform
+          make OS=darwin ARCH=amd64 release/terraform release/event-handler
+          make OS=darwin ARCH=arm64 release/terraform
+          make OS=darwin ARCH=universal release/terraform
+      - name: Collect the build files
+        run: |
+          mkdir -pv "$ARTIFACT_DIRECTORY"
+          cp $(find . '(' -name "*.tar.gz" -o -name "*.tgz" ')' -type f) "$ARTIFACT_DIRECTORY/"
+      - name: Generate checksum files for built files
+        working-directory: ${{ env.ARTIFACT_DIRECTORY }}
+        run: |
+          find . '(' -name "*.tar.gz" -o -name "*.tgz" ')' -type f -exec sh -c 'sha256sum "$(basename {})" > "{}.sha256"' \;
+          echo "Artifacts:"
+          ls -lh
+      - name: Assume AWS role for uploading the artifacts
+        uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+        with:
+          role-skip-session-tagging: true
+          aws-region: us-west-2
+          role-to-assume: ${{ vars.ARTIFACT_UPLOAD_AWS_ROLE }}
+          role-session-name: "artifact-upload-${{ github.run_number }}"
```

Comment:
> Perhaps this needs a repo in it too

I think this is available via Cloudtrail logs and therefore not needed here. @wadells if you have time could you confirm if the repo name is available via Cloudtrail? Also, do you/security have any preferences or requirements on what the RSN should be? My understanding is that the value of this field is ultimately to help Security team trace log events back to specific workflow invocations/steps.

Either way, I will update this to a standard - just want to know if `<workflow>-[step]-${{ github.run_attempt }}` is the best route or if there is another one that we should follow.

Reply:
> > FYI, I've adopted an unvoiced convention of naming the sessions <workflow>-[step]-${{ github.run_attempt }} since the purpose of the session name is to track it back to what did it. Perhaps this needs a repo in it too

We don't currently record information about the step. We definitely record information about the repo and environment.  Here's an example GHA role assumption, pulled from our SIEM.

```
{
	"awsRegion": "us-west-2",
	"eventCategory": "Management",
	"eventID": "b4a24080-0dd3-4191-91de-774907777b46",
	"eventName": "AssumeRoleWithWebIdentity",
	"eventSource": "sts.amazonaws.com",
	"eventTime": "2024-02-05 01:04:59",
	"eventType": "AwsApiCall",
	"eventVersion": "1.08",
	"managementEvent": true,
	"p_any_actor_ids": [
		"arn:aws:iam::146628656107:oidc-provider/token.actions.githubusercontent.com/teleport:sts.amazonaws.com:repo:gravitational/cloud:environment:renovate"
	],
	"p_any_aws_account_ids": [
		"146628656107"
	],
	"p_any_aws_arns": [
		"arn:aws:iam::146628656107:role/cloud-github-action-renovate-role",
		"arn:aws:sts::146628656107:assumed-role/cloud-github-action-renovate-role/GitHubActions",
		"arn:aws:iam::146628656107:oidc-provider/token.actions.githubusercontent.com/teleport"
	],
	"p_any_ip_addresses": [
		"40.84.174.103"
	],
	"p_any_trace_ids": [
		"ASIASEI57XPV7BZK6H3J"
	],
	"p_any_usernames": [
		"repo:gravitational/cloud:environment:renovate"
	],
	"p_event_time": "2024-02-05 01:04:59",
	"p_log_type": "AWS.CloudTrail",
	"p_parse_time": "2024-02-05 01:20:25.847",
	"p_row_id": "4e57ac904eab8dbbd6dff1b11df3af02",
	"p_schema_version": 0,
	"p_source_file": {
		"aws_s3_bucket": "aws-controltower-logs-703521745945-us-west-2",
		"aws_s3_key": "AWSLogs/o-0p9sx5fohd/146628656107/CloudTrail/us-west-2/2024/02/05/146628656107_CloudTrail_us-west-2_20240205T0110Z_8JGuTW5RKzxJY9qi.json.gz"
	},
	"p_source_id": "84888de7-4fc0-4ea1-94de-55ab38080669",
	"p_source_label": "gravitational-organization-logs",
	"p_udm": {
		"source": {
			"address": "40.84.174.103",
			"ip": "40.84.174.103"
		},
		"user": {
			"arns": [
				"arn:aws:iam::146628656107:role/cloud-github-action-renovate-role",
				"arn:aws:sts::146628656107:assumed-role/cloud-github-action-renovate-role/GitHubActions"
			],
			"name": "repo:gravitational/cloud:environment:renovate"
		}
	},
	"readOnly": true,
	"recipientAccountId": "146628656107",
	"requestID": "0e4b885b-c20e-4e9f-bf4b-548980637049",
	"requestParameters": {
		"durationSeconds": 3600,
		"roleArn": "arn:aws:iam::146628656107:role/cloud-github-action-renovate-role",
		"roleSessionName": "GitHubActions"
	},
	"resources": [
		{
			"accountId": "146628656107",
			"arn": "arn:aws:iam::146628656107:role/cloud-github-action-renovate-role",
			"type": "AWS::IAM::Role"
		}
	],
	"responseElements": {
		"assumedRoleUser": {
			"arn": "arn:aws:sts::146628656107:assumed-role/cloud-github-action-renovate-role/GitHubActions",
			"assumedRoleId": "AROASEI57XPV5ASV6JUYK:GitHubActions"
		},
		"audience": "sts.amazonaws.com",
		"credentials": {
			"accessKeyId": "ASIASEI57XPV7BZK6H3J",
			"expiration": "Feb 5, 2024, 2:04:59 AM",
			"sessionToken": "<redacted>"
		},
		"provider": "arn:aws:iam::146628656107:oidc-provider/token.actions.githubusercontent.com/teleport",
		"subjectFromWebIdentityToken": "repo:gravitational/cloud:environment:renovate"
	},
	"sourceIPAddress": "40.84.174.103",
	"tlsDetails": {
		"cipherSuite": "TLS_AES_128_GCM_SHA256",
		"clientProvidedHostHeader": "sts.us-west-2.amazonaws.com",
		"tlsVersion": "TLSv1.3"
	},
	"userAgent": "aws-sdk-js/3.423.0 ua/2.0 os/linux#6.2.0-1019-azure lang/js md/nodejs#20.8.1 api/sts#3.423.0 configure-aws-credentials-for-github-actions",
	"userIdentity": {
		"identityProvider": "arn:aws:iam::146628656107:oidc-provider/token.actions.githubusercontent.com/teleport",
		"principalId": "arn:aws:iam::146628656107:oidc-provider/token.actions.githubusercontent.com/teleport:sts.amazonaws.com:repo:gravitational/cloud:environment:renovate",
		"type": "WebIdentityUser",
		"userName": "repo:gravitational/cloud:environment:renovate"
	}
}
```

Reply:
> do you/security have any preferences or requirements on what the RSN should be? My understanding is that the value of this field is ultimately to help Security team trace log events back to specific workflow invocations/steps.

`<workflow>-[step]-${{ github.run_attempt }}` seems good to me.

Reply:
Thanks for posting that sample json - from that we can tell it is github actions and the cloud repo, but not which workflow or step. I had prefixed the session-role-name with `gha` in some cases, but I'll stop doing that since it's clear it is github, and I was considering whether the repo was needed but also not needed it seems. What is missing is the branch, since the workflows can be different on different branches, but the timestamps should be enough to match up one of these events with a workflow run and the branch/tag will be clear from there.